Java/Core/src/main/cpp/jni/ethereum/com_breadwallet_core_ethereum_BREthereumEWM.c

```diff
@@ -1796,21 +1786,19 @@ clientGetNonce(BREthereumClientContext context,
 static void
 clientEWMEventHandler(BREthereumClientContext context,
                       BREthereumEWM ewm,
-                      BREthereumEWMEvent event,
-                      BREthereumStatus status,
-                      const char *errorDescription) {
+                      BREthereumEWMEvent event) {
     JNIEnv *env = getEnv();
     if (NULL == env) return;
 
-    jstring errorDescriptionString = (NULL == errorDescription
+    jstring errorDescriptionString = ('\0' == event.errorDescription[0]
```

Comment:
I guess this is the safest, most direct, approach.  Can't meaningfully compare SUCCESS and event.type.

Reply:
I sort of assumed you were setting the description to an empty string but is that not the case? This could be problematic if not. Should this be a check on if `SUCCESS == event.status`?

Reply:
I don't think `errorDescription` is set consistently in the cases where the status is not `SUCCESS`.